src/utils/appinsights.ts

```diff
@@ -0,0 +1,23 @@
+import * as appInsights from "applicationinsights";
+export class AppInsightsClientBuilder {
+  private client: appInsights.TelemetryClient;
+  constructor(instrumentalKey: string) {
+    appInsights
+      .setup(instrumentalKey)
+      .setAutoDependencyCorrelation(true)
+      .setAutoCollectRequests(false)
```

Comment:
curious why? it doesn't work?

Reply:
I've preferred the manual collection of app requests to filter query parameters and avoid to sent that data to App Insights.

Reply:
You may consider `client.trackNodeHttpRequest()` if you want to skip the time diff:

https://github.com/microsoft/ApplicationInsights-node.js#examples

it should accept properties (ie. `success`).

Reply:
@BurnedMarshal appinsight supports [telemetry processors](https://github.com/microsoft/ApplicationInsights-node.js/blob/9d25949f5cdac42289b41b05039cf400e5b80733/README.md#preprocess-data-with-telemetry-processors), kind of middlewares that can preprocess the data before it gets sent to Azure - I suggest do use that facility as it makes the code cleaner and simpler

Reply:
@gunzip ok, I've refactored the code to use `trackNodeHttpRequest()`. @cloudify Now to filter the query params sent to app insights is used the telemetry preprocessor. Fixed in 41c9e35

Reply:
@BurnedMarshal it's not clear to me why we're manually tracking requests with `trackNodeHttpRequest` instead of using `setAutoCollectRequests(true)`

Reply:
@cloudify effectively now with the telemetry preprocessor we could use `setAutoCollectRequests(true)`. To avoid data duplication `trackNodeHttpRequest` must be removed. Actually is more explicit and user-readable that when the HTTP server is created something could happen. If you prefer we can change this approach to use automatic data collection.

Reply:
yes that's a good point, I think we can solve the issue with a BIG comment describing what's going on - since this is more an "instrumentation" side effect that doesn't change the behavior of the app, I would go for the solution that minimizes the code

Reply:
Fixed in 5d56f09